src/trezor.rs

```diff
@@ -0,0 +1,185 @@
+//! # Examples
+//! ```no_run
+//! use async_hwi::trezor::TrezorClient;
+//! use async_hwi::HWI;
+//!
+//! #[tokio::main]
+//! pub async fn main() {
+//!     let mut hwi = TrezorClient::connect_first(false).unwrap();
+//!     hwi.set_network(bitcoin::Network::Bitcoin);
+//!     println!("{}", hwi.get_version().await.unwrap());
+//!     println!("{:?}", hwi.get_master_fingerprint().await);
+//!     let path =
+//!         <bitcoin::bip32::DerivationPath as std::str::FromStr>::from_str("m/44'/1'/0'/0/0")
+//!             .expect("Failed to parse path");
+//!     println!("{:?}", hwi.get_extended_pubkey(&path).await);
+//!}
+//!  ```
+
+use std::{cell::RefCell, str::FromStr};
+
+use async_trait::async_trait;
+use bitcoin::{
+    bip32::{DerivationPath, ExtendedPubKey, Fingerprint},
+    psbt::Psbt,
+};
+use trezor_client::{Trezor, TrezorResponse};
+
+use crate::{DeviceKind, Error, HWI};
+
+pub struct TrezorClient {
+    client: RefCell<Trezor>,
+    kind: DeviceKind,
+    network: bitcoin::Network,
+}
+
+unsafe impl Sync for TrezorClient {}
```

Comment:
Can you comment why do you need unsafe here ? If it is because of the async threads, maybe it would be judicious to use std::sync::Arc instead of RefCell, or maybe directly embed the Trezor in the TrezorClient struct.

Reply:
I couldn't directly embed the Trezor in TrezorClient because the HWI trait functions specify the self reference to be immutable, but some of Trezor's functions need it to be mutable, thus the smart pointers are needed.  Regarding avoiding the unsafe impl, I will get back to you.